python/semantic_kernel/orchestration/sk_function.py

```diff
@@ -87,9 +91,11 @@ def from_native_method(method, skill_name="", log=None) -> "SKFunction":
             and method.__sk_function_input_description__ != ""
         ):
             input_param = ParameterView(
-                "input",
-                method.__sk_function_input_description__,
-                method.__sk_function_input_default_value__,
+                name="input",
+                description=method.__sk_function_input_description__,
+                default_value=method.__sk_function_input_default_value__,
+                type="string",
+                required=True,
```

Comment:
@awharrison-28 also commented on this, and we figured it makes sense to set all parameters as required for function calling, instead of setting it to default False and thereby inviting more cases where GPT doesn't respond with a value for that parameter. We might want to consider renaming this field to something like function_call_required to clarify that is is used only for function calling.

Reply:
There's overlap here with needs of planner, as well. So far, implementations have relied on dictating whether a value is required in the description of the parameter -- hacky but sufficient in a semantic world. Whatever changes we make should be aligned across these scenarios.

Reply:
For semantic functions, all parameters are technically optional even if they don't have a default value. For native functions, we can be smarted to determine requiredness.